core/src/main/scala/com/codacy/analysis/core/tools/Tool.scala

```diff
@@ -128,15 +140,7 @@ class Tool(runner: ToolRunner, defaultRunTimeout: Duration)(private val tool: To
 object Tool {
 
   def apply(fullToolSpec: FullToolSpec, languageToRun: Language): Tool = {
-    val dockerInformation = new DockerInformation(
-      dockerImageName = fullToolSpec.tool.dockerImage,
-      needsCompilation = fullToolSpec.tool.needsCompilation)
-
-    val dockerRunner = new BinaryDockerRunner[Result](dockerInformation)
-
-    val runner =
-      new ToolRunner(Option(fullToolSpec.toolApiSpec), fullToolSpec.tool.prefix, dockerRunner)
-    new Tool(runner, DockerRunner.defaultRunTimeout)(fullToolSpec.tool, languageToRun)
```

Comment:
Differently from metrics and duplication, the issues tools were defining the `dockerRunner` during instantiation instead of during the `run` method. I changed here to be consistent with metrics and duplication so I can easily pass the memory configuration together with other run configurations.